File: 5-Patron/php/HabilidadMagica.php
<?php
abstract class HabilidadMagica{
	private $nivel;
	private $descripcion;

	public function __construct($nivel, $descripcion){
    $this->nivel = $nivel;
    $this->descripcion = $descripcion;
	}

	public function getNivel(){
		return $this->nivel;	
	}

	public function setNivel($nivel){
		$this->nivel = $nivel;
	}

	public function getDescripcion(){
		return $this->descripcion;	
	}

	public function setDescripcion($descripcion){
		$this->descripcion = $descripcion;
	}

	public function verInfoHabilidad(){
    return  "Nivel: " . $this->nivel .
            "Descripción: " . $this->descripcion;
  	}

  	public function  subirNivelHabilidad(){
  		$this->nivel +=1;
  	}
}
?>

File: 5-Patron/Java/HabilidadMagica.java
package patron5;

public class HabilidadMagica {
    private int nivel;
    private String descripcion;

    public HabilidadMagica(int nivel, String descripcion) {
        this.nivel = nivel;
        this.descripcion = descripcion;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public int getNivel() {
        return nivel;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public void setNivel(int nivel) {
        this.nivel = nivel;
    }
    
    public String verInfoHabilidad(){
        return ("Nivel: "+ this.nivel+ "\nDescripcion: "+this.descripcion);
    }
    public void subirNivelHabilidad(){
        this.nivel +=1;
    }
}
